Sources/Sentry/SentryCurrentDate.m

```diff
@@ -0,0 +1,33 @@
+#import <Foundation/Foundation.h>
+#import "SentryCurrentDate.h"
+#import "SentryCurrentDateProvider.h"
+#import "SentryDefaultCurrentDateProvider.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SentryCurrentDate ()
+
+@end
+
+@implementation SentryCurrentDate
+
+static id<SentryCurrentDateProvider> currentDateProvider;
+
++(NSDate* _Nonnull) date {
+    @synchronized(self) {
```

Comment:
Does it make sense to we use a double-check lock here? This way we don't create contention once the field is initialized:

```suggestion
    if (nil != currentDateProvider) {
        return [currentDateProvider date];
    }
    @synchronized(self) {
```